### ISSUE TITLE
Support of TypeToken

### ISSUE THREAD

**Issue by m0mus** [eclipse-ee4j/jsonb-api#165]
Extend deserialize family with a method accepting `TypeToken`. It's more convenient and user friendly than using pure `Type`. Similar mechanism is used in Gson and Jackson.

Deserialization of generic list as it is now:

```
// List of dogs
List<Dog> dogs = new ArrayList<>();

// Initialization
...

// Create Jsonb and deserialize
Jsonb jsonb = JsonbBuilder.create();
dogs = jsonb.fromJson(result, new ArrayList<Dog>(){}.getClass().getGenericSuperclass());
```
Deserialization of generic list using TypeToken:

```
// List of dogs
List<Dog> dogs = new ArrayList<>();

// Initialization
...

// Create Jsonb and deserialize
Jsonb jsonb = JsonbBuilder.create();
dogs = jsonb.fromJson(result, TypeToken<List<Dog>>(){}.getType());
```

**Comment by rmannibucau** [comment]
Do you have the related discussion of the spec? Was rejected cause already too much duplicated accross the platform (see jaxrs and cdi to cite a few).
Jakata.commons.next sounds a saner place IMHO.

**Comment by m0mus** [comment]
If there were such discussions I didn't participate there. I agree that ideally it should be in JDK or in Common Annotations. On the other hand I don't think that other specs which use their own token solutions will switch to it.

**Comment by rmannibucau** [comment]
Jaxrs and cdi would support it - cdi already btw, they didnt type anything with their typetoken and just used Type or Annotation as us. Jaxrs used it but can now move on so no need to just reinvent the wheel and let's either benefit from the donation or just stick to current option which works not that bad at all.

**Comment by m0mus** [comment]
@rmannibucau I disagree that current option works not bad at all. Do you want to take an initiative and propose common platform level TypeToken to Jakarta EE Platform?

**Comment by rmannibucau** [comment]
Maybe a compromise might be:

> dogs = jsonb.fromJson(result, jsonb.createParameterizedType(List.class, Dog.class));
> dogs = jsonb.fromJson(result, jsonb.createParameterizedType(Map.class, String.class, Dog.class));
> dogs = jsonb.fromJson(result, jsonb.createParameterizedType(Map.class, String.class, jsonb.createParameterizedType(List.class, Animal.class)));
